src/_pytest/python.py

```diff
@@ -486,11 +503,24 @@ def _genfunctions(self, name: str, funcobj) -> Iterator["Function"]:
             module=module,
             _ispytest=True,
         )
-        methods = []
+        methods = [unwrap_metafunc_parametrize_and_possibly_prune_dependency_tree]
         if hasattr(module, "pytest_generate_tests"):
             methods.append(module.pytest_generate_tests)
         if cls is not None and hasattr(cls, "pytest_generate_tests"):
             methods.append(cls().pytest_generate_tests)
+
+        setattr(metafunc, "has_dynamic_parametrize", False)
+
+        @wraps(metafunc.parametrize)
```

Comment:
Gotta say I do not really like this magic here.

Reply:
I changed the solution to one that does `if metafunc._calls` but this does not differentiate between direct and indirect dynamic parametrization. It's OK but not perfect regarding being optimized.
 
I note that I had forgotten to differentiate between the two in the former solution, but I could do it therein as opposed to the new solution.